client/web/src/auth/SignInSignUpCommon.tsx

```diff
@@ -76,3 +76,15 @@ export function getReturnTo(location: H.Location): string {
     newURL.searchParams.append('toast', 'integrations')
     return newURL.pathname + newURL.search + newURL.hash
 }
+
+export function maybeAddPostSignUpRedirect(url?: string): string {
+    const enablePostSignupFlow = window.context?.experimentalFeatures?.enablePostSignupFlow
+    const isDotCom = window.context?.sourcegraphDotComMode
+    const shouldAddRedirect = isDotCom && enablePostSignupFlow
+
+    if (url) {
+        return shouldAddRedirect ? `${url}&redirect=/welcome` : url
```

Comment:
If `url` does not have a query string already, this would fail. Should we protect ourselves against that? E.g. by using the `URLSearchParams` object instead?
```suggestion
        const urlObject = new URL(url, 'https://foo/') // second param to protect against relative urls
        if (shouldAddRedirect) {
          urlObject.searchParams.append('redirect', '/welcome');
        }
        return urlObject.toString();
```

Reply:
Nice catch, thanks!